src/Admin/Input/Integer.php

```diff
@@ -20,6 +20,15 @@ public function __construct() {
 		// the text field block to work around it.
 		parent::__construct( 'integer', 'woocommerce/product-text-field' );
 
-		$this->set_block_attribute( 'type', [ 'value' => 'number' ] );
+		$this->set_block_attribute(
+			'pattern',
+			[
+				'value'   => '\d+',
```

Comment:
Maybe the pattern could be `[0-9]\d?` to avoid saving a value like '000123'.

Reply:
Not sure about this. If I use `[0-9]\d`, it will only match two digits and won't work for single-digit numbers. If I switch to `[0-9]\d+`, it will match any number of digits but won't prevent cases like `000123`. To handle that, I think the pattern needs to be something like `^(0|[1-9]\d*)$`. WDYT?

Reply:
Sorry, it should be `[0-9]\d*`.

> To handle that, I think the pattern needs to be something like `^(0|[1-9]\d*)$`. WDYT?

`^(?:` and `)$` impliedly wrap up the pattern of `<input>`, so it will match the entire value rather than matching a substring.

Reply:
mmm, I tried using `[0-9]\d*` but I got a similar result to `0-9]\d+`, we are still able to type `00123`.

Reply:
Sorry again 😂. I just tested with `\d|[1-9]\d*` and hope it's correct this time.

Reply:
I was too focused on the `^` and `$` parts. Just noticed the pattern `0|[1-9]\d*` you commented in https://github.com/woocommerce/google-listings-and-ads/pull/2511#discussion_r1715375981 should work as well.

Reply:
Adjusted in https://github.com/woocommerce/google-listings-and-ads/pull/2511/commits/ea781f60c49d9b37ec68c3d4e98b1ad6cf56085e with more assertions for invalid values.